### PR TITLE
[Java] Added x-obfuscated for toString() support

### PR DESCRIPTION
**Pull request by rybchinsky**
Added ability to hide personal data when toString for class is calling.
Does not work in cases only one field is printed.

### CODE REVIEW

File: test-schemas/src/main/resources/openapi/simple.yaml [hunk 1]
```diff
@@ -97,6 +97,11 @@ components:
   schemas:
     dto:
       type: object
+      x-obfuscated:
```
Comment (andrey-fomin) [inline test-schemas/src/main/resources/openapi/simple.yaml]: I believe it's better to move this property at value-type level.
Also this feature is very specific to languages like java with method `toString`. For example, it's not applicable to TypeScript.

Maybe it's better to implement this feature like this

```yaml
type: object:
properties:
  property1:
    type: string
    x-to-string-obfuscated: true
```
```

Comment (rybchinsky) [reply test-schemas/src/main/resources/openapi/simple.yaml]: 1) i tried to implement both ways (class-type and value-type), as for required, but had no success (no kotlin experience before);
i agree to create debt on it

2) name was from https://github.com/swagger-api/swagger-codegen/pull/7060 ; 
don't think its good idea to create another alias for this feature

Comment (andrey-fomin) [reply test-schemas/src/main/resources/openapi/simple.yaml]: ok, it's better to implement it as it's already implemented in swagger-codegen

Comment (rybchinsky) [reply test-schemas/src/main/resources/openapi/simple.yaml]: first comment is fixed too

File: core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt [hunk 2]
```diff
@@ -59,6 +59,11 @@ class JsonSchema(override val fragment: Fragment, override val parent: TypedFrag
         null -> emptySet()
         else -> requiredFragment.map { propertyNameFragment -> propertyNameFragment.asString() }.toSet()
     }
+
+    fun obfuscated() = when (val obfuscatedFragment = fragment.getOptional("x-obfuscated")) {
+        null -> emptySet()
```
Comment (andrey-fomin) [inline core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: why Set?

File: test-schemas/src/main/resources/openapi/simple.yaml [hunk 3]
```diff
@@ -100,3 +100,7 @@ components:
       properties:
         property1:
           type: string
+          required: true
+          x-obfuscated: true
```
Comment (andrey-fomin) [inline test-schemas/src/main/resources/openapi/simple.yaml]: This file and related tests should not be used for testing dto-specific logic.
Testing should be done in io.github.fomin.oasgen.java.dto.jackson.wstatic.BuiltinConverterMatchersTest.

File: core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt (hunk 2 above)
Comment (rybchinsky) [reply core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: remove duplications, the same logic as for `required` above

File: test-schemas/src/main/resources/openapi/simple.yaml (hunk 3 above)
Comment (rybchinsky) [reply test-schemas/src/main/resources/openapi/simple.yaml]: dto-specific param is also processed, i update this file and add test to ObjectConverterMatcherTest

Comment (rybchinsky) [reply test-schemas/src/main/resources/openapi/simple.yaml]: done; here we have dto-specific logic; property-specific test added to ObjectConverterMatcherTest

File: core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt [hunk 4]
```diff
@@ -86,3 +86,15 @@ fun JsonSchema.jointProperties(): Map<String, JsonSchema> = jointPropertiesList(
             }
             acc
         }
+
+fun JsonSchema.propertyObfuscated(): List<String> {
```
Comment (andrey-fomin) [inline core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: `obfruscatedProperties` probably a better name

Comment (andrey-fomin) [inline core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: `obfruscatedProperties` probably a better name

Comment (rybchinsky) [reply core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: fixed

Comment (rybchinsky) [reply core/src/main/kotlin/io/github/fomin/oasgen/JsonSchema.kt]: duplicated, already fixed

**Pull request merged**